src/test/java/HorseTest.java

```diff
@@ -0,0 +1,114 @@
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.mockito.MockedStatic;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.mockStatic;
+
+class HorseTest {
+
+    @Test
+    public void constructor_NullNameParam_throwsIllegalArgumentException(){
+        Throwable exception = assertThrows(IllegalArgumentException.class, () -> new Horse(null, 1,2));
+        assertEquals("Name cannot be null.", exception.getMessage());
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {" ", "  ", "\n", "\n\n","\t","\t\t","\t \t","\r"})
+    public void constructor_EmptyNameParam_throwsIllegalArgumentException(String name){
+        String expectedMessage = "Name cannot be blank.";
+
+        Throwable exception = assertThrows(IllegalArgumentException.class, () -> new Horse(name, 1,2));
+        assertEquals(expectedMessage, exception.getMessage());
+    }
+
+    @Test
+    public void constructor_NegativeSpeedParamPassed_throwsIllegalArgumentException(){
+        String expectedMessage = "Speed cannot be negative.";
+        String name = "TestName";
+        double speed = -3;
+        double distance = 2;
+
+        Throwable exception = assertThrows(IllegalArgumentException.class, () -> new Horse(name, speed,distance));
+        assertEquals(expectedMessage, exception.getMessage());
+    }
+
+    @Test
+    public void constructor_NegativeDistanceParamPassed_throwsIllegalArgumentException(){
+        String expectedMessage = "Distance cannot be negative.";
+        String name = "TestName";
+        double speed = 1;
+        double distance = -2;
+
+        Throwable exception = assertThrows(IllegalArgumentException.class, () -> new Horse(name, speed,distance));
+        assertEquals(expectedMessage, exception.getMessage());
+    }
+
+    @Test
+    void getName_ReturnsCorrectName() {
+        String name = "TestName";
+        double speed = 1;
+        double distance = 2;
+        Horse horse = new Horse(name, speed,distance);
+
+        String actualName = horse.getName();
+
+        assertEquals(name, actualName);
+    }
+
+    @Test
+    void getSpeed_ReturnsCorrectSpeed() {
+        String name = "TestName";
+        double speed = 1;
```

Comment:
u are duplacate this variables on all tests, just create all in class level and reuse it